discovery/gossiper_test.go

```diff
@@ -3353,13 +3353,23 @@ func TestPropagateChanPolicyUpdate(t *testing.T) {
 
 	const numChannels = 3
 	channelsToAnnounce := make([]*annBatch, 0, numChannels)
+	channelPoints := make([]OptionalMsgField, 0, numChannels)
 	for i := 0; i < numChannels; i++ {
 		newChan, err := createAnnouncements(uint32(i + 1))
 		if err != nil {
 			t.Fatalf("unable to make new channel ann: %v", err)
 		}
 
 		channelsToAnnounce = append(channelsToAnnounce, newChan)
+
+		// channelPoints ensures that each channel policy in the map
+		// returned by PropagateChanPolicyUpdate has a unique key. Since
+		// the map is keyed by wire.OutPoint, we want to ensure that
+		// each channel has a unique channel point.
+		channelPoints = append(
+			channelPoints,
+			ChannelPoint(wire.OutPoint{Index: uint32(i)}),
+		)
```

Comment:
You could also not build the slice and just create the outpoint inside the `channelsToAnnounce` loop below.